cmd/lotus/daemon.go

```diff
@@ -574,3 +600,88 @@ func ImportChain(ctx context.Context, r repo.Repo, fname string, snapshot bool)
 
 	return nil
 }
+
+func slashConsensus(a lapi.FullNode, from string, extra string) error {
+	ctx := context.Background()
+
+	blocks, err := a.SyncIncomingBlocks(ctx)
+	if err != nil {
+		return err
+	}
+
+	for block := range blocks {
+		log.Infof("deal with block: %d, %v, %s", block.Height, block.Miner, block.Cid())
+		if otherBlock, err := a.SlashFilterMinedBlock(ctx, block); err != nil {
+			if otherBlock == nil {
+				continue
+			}
+			log.Errorf("<!!> SLASH FILTER ERROR: %s", err)
+			var fromAddr address.Address
+			if from == "" {
```

Comment:
Let's set the sender address at the start of this method (outside of the for loop).